lib/ddtrace/transport/http.rb

```diff
@@ -75,13 +76,27 @@ def default_headers
       end
 
       def default_hostname
+        return default_url.hostname if default_url
+
         ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_HOST, Datadog::Ext::Transport::HTTP::DEFAULT_HOST)
       end
 
       def default_port
+        return default_url.port if default_url
+
         ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_PORT, Datadog::Ext::Transport::HTTP::DEFAULT_PORT).to_i
       end
 
+      def default_url
+        url_env = ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_URL, nil)
+
+        if url_env
+          uri_parsed = URI.parse(url_env)
+
+          uri_parsed if %w[http https].include?(uri_parsed.scheme)
```

Comment:
is `https` honored anywhere? if I set `https://127.0.0.1:8106/` will that actually use TLS?

Reply:
I ...don't think so? mainly I just wanted to avoid a malformed url being passed in